asdf/commands/tests/test_edit.py

```diff
@@ -0,0 +1,231 @@
+            if "foo" in l:
+                print("foo: 41", file=fd)  #  Change a value
+            else:
+                fd.write(l)
+
+    return oname
+
+
+def create_edit_smaller(base_yaml):
+    """
+    The YAML from the base ASDF file will have a 'foo' value of 42.  Create
+    an edited YAML file with this value being 41.  This will create an edited
+    YAML file with the same number of characters in the YAML section as was in
+    the original ASDF file.
+    """
+    with open(base_yaml, "r") as fd:
+        lines = fd.readlines()
+
+    base, ext = os.path.splitext(base_yaml)
+    oname = f"{base}_edit_smaller.yaml"
+    if os.path.exists(oname):
+        os.remove(oname)
+    with open(oname, "w") as fd:
+        for l in lines:
+            if "foo" in l:
+                print("foo: 2", file=fd)  #  Change a value
+            else:
+                fd.write(l)
+
+    return oname
+
+
+def create_edit_larger(base_yaml):
+    """
+    The YAML from the base ASDF file will have a 'foo' value.  After this
+    line, add another line.  This will create an edited YAML file that will
+    have more characters than the YAML portion of the original ASDF file.
+    """
+    with open(base_yaml, "r") as fd:
+        lines = fd.readlines()
+
+    base, ext = os.path.splitext(base_yaml)
+    oname = f"{base}_edit_larger.yaml"
+    if os.path.exists(oname):
+        os.remove(oname)
+    with open(oname, "w") as fd:
+        for l in lines:
+            fd.write(l)
+            if "foo" in l:
+                print("bar: 13", file=fd)  # Add a line
+
+    return oname
+
+
+def copy_base_asdf_equal(base_asdf):
+    """
+    Create an ASDF file from the base ASDF file to test the editing of the
+    YAML portion with equal number of YAML characters.
+    """
+    base, ext = os.path.splitext(base_asdf)
+    oname = f"{base}_equal.asdf"
+    if os.path.exists(oname):
+        os.remove(oname)
+    shutil.copyfile(base_asdf, oname)
+
+    return oname
+
+
+def copy_base_asdf_smaller(base_asdf):
+    """
+    Create an ASDF file from the base ASDF file to test the editing of the
+    YAML portion with equal number of YAML characters.
+    """
+    base, ext = os.path.splitext(base_asdf)
+    oname = f"{base}_smaller.asdf"
+    if os.path.exists(oname):
+        os.remove(oname)
+    shutil.copyfile(base_asdf, oname)
+
+    return oname
+
+
+def copy_base_asdf_larger(base_asdf):
+    """
+    Create an ASDF file from the base ASDF file to test the editing of the
+    YAML portion with a larger number of YAML characters.
+    """
+    base, ext = os.path.splitext(base_asdf)
+    oname = f"{base}_larger.asdf"
+    if os.path.exists(oname):
+        os.remove(oname)
+    shutil.copyfile(base_asdf, oname)
+
+    return oname
+
+
+@pytest.mark.parametrize("version", asdf.versioning.supported_versions)
+def test_edits(tmpdir,version):
+    #        Test:
+    # Create base ASDF file for testing
+    tmpdir = "/Users/kmacdonald/tmp"
```

Comment:
Looks like debris left over from local testing?

Reply:
This has been removed.  It was for local testing and debugging.